### PR TITLE
feat(Data Grid): align resize coulum PAL pattern

### PR DESCRIPTION
**Pull request by Ratheeshrajan**
Contributes to #1923

**Resize multiple columns at a time :**

- Hold Shift, and hover over the right side of any column header until a horizontal split cursor appears. Click on the resizer to select the columns. 
 image

- Double-click the column resizer to resize all visible columns to fit the content.
image

- Display minimum of one letter in a column
image
- Wrap column header titles
image

#### What did you change?
Datagrid/DatagridHeaderRow.js
Datagrid/DatagridRow.js
Datagrid/Datagrid.stories.js
styles/_datagrid.scss

#### How did you test and verify your work?
Storybook

Keyboard navigation will be covered in another PR later.

### CODE REVIEW

**Comment by Ratheeshrajan** [comment]
V-11 PR added https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/2352

**Comment by matthewgallo** [comment]
Hey @Ratheeshrajan, I'm noticing an issue where if I resize multiple columns then try to resize a single column, I'm not able to resize the single column anymore.

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/DatagridHeaderRow.js [hunk 1]
```diff
@@ -6,56 +6,186 @@
...
-                className={`${blockClass}__resizer`}
-              />
-            )}
-          </TableHeader>
+const HeaderRow = (datagridState, headRef, headerGroup) => {
+  const [selectedHeader, setSelectedHeader] = useState([]);
+  const [selectedHeaderWidths, setSelectedHeaderWidths] = useState([]);
+  const [isResizing, setIsResizing] = useState('');
+  const [isDblClick, setIsDblClick] = useState(false);
+  const [dargStopped, setDargStopped] = useState(false);
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/DatagridHeaderRow.js:22]: Typo `darg` => `drag`

**Review by matthewgallo — CHANGES_REQUESTED** [review]
I'm noticing an issue where if I resize multiple columns then try to resize a single column, I'm not able to resize the single column anymore. I'm also noticing the same behavior after the double click interaction on a column, not able to resize that column anymore afterwards.

**Comment by AlexanderMelox** [comment]
Weird bug when resizing the first column to the smallest width
https://user-images.githubusercontent.com/12755042/230482191-0d6110f3-bc0a-4a2e-a9cf-2b7ab40fbd2e.mov

**Comment by elycheea** [comment]
@Ratheeshrajan Looks like there’s a conflict in the SCSS still!

**Comment by Ratheeshrajan** [comment]
> @Ratheeshrajan Weird bug when resizing columns to the smallest width https://user-images.githubusercontent.com/12755042/230482191-0d6110f3-bc0a-4a2e-a9cf-2b7ab40fbd2e.mov

@AlexanderMelox It's an existing issue width React table `state.columnResizing` event. It's not stopping the resize event even if reached `width:0` and it's goes to `-` value in `columnResizing.columnWidth` and then resets the column width to the column `originalWidth` 150px, also it's ignoring the min-widths as well. This happens if we resize very fast. I have tried a workaround that I can provide this time, if the column width goes beyond the min-width and rests back to the original width then can rest the column width to a min-width 50. The issue with this approach is, this will only take effect after the Reat table `columnResizing` ends. So we can see a slight flickering when we are resizing very fast. This issue is reproducible in Carbon for IBM Storybook.  

Issue in https://carbon-for-ibm-products.netlify.app/:
https://github.com/carbon-design-system/ibm-cloud-cognitive/assets/305492/aee54cc1-1f76-4986-9993-2f05c4142a05

Workaround:

https://github.com/carbon-design-system/ibm-cloud-cognitive/assets/305492/b73bc794-2cda-4b37-a5dc-f0526578985f

**Comment by matthewgallo** [comment]
Closing since there have been several updates to resizing since this PR was initially opened. But for documentation purposes, our current approach utilizes the `stateReducer` that `react-table` supports and that's how we've been able to hook into updating `datagridState.state`. By dispatching actions in this reducer and providing this reducer to the `useTable` hook, those `state` values are always synched with the resized value.

**Pull request closed without merging**